ibtk/src/lagrangian/FEDataManager.cpp

```diff
@@ -1270,17 +1302,34 @@ FEDataManager::interpWeighted(const int f_data_idx,
     else
     {
         // Extract local form vectors.
+        int ierr;
         auto X_petsc_vec = static_cast<PetscVector<double>*>(&X_vec);
         Vec X_global_vec = X_petsc_vec->vec();
         Vec X_local_vec;
-        VecGhostGetLocalForm(X_global_vec, &X_local_vec);
+        ierr = VecGhostGetLocalForm(X_global_vec, &X_local_vec);
+        IBTK_CHKERRQ(ierr);
         double* X_local_soln;
-        VecGetArray(X_local_vec, &X_local_soln);
+        ierr = VecGetArray(X_local_vec, &X_local_soln);
+        IBTK_CHKERRQ(ierr);
+
+        auto F_petsc_vec = static_cast<PetscVector<double>*>(&F_vec);
+        const bool ghosted_vec = F_petsc_vec->type() == GHOSTED;
+        Vec F_global_vec = F_petsc_vec->vec();
+        Vec F_local_vec;
+        double* F_local_soln;
+        if (ghosted_vec)
+        {
+            ierr = VecGhostGetLocalForm(F_global_vec, &F_local_vec);
+            IBTK_CHKERRQ(ierr);
+            ierr = VecZeroEntries(F_local_vec);
+            IBTK_CHKERRQ(ierr);
+            ierr = VecGetArray(F_local_vec, &F_local_soln);
+            IBTK_CHKERRQ(ierr);
+        }
```

Comment:
I think that this and the indexing are really all we need.

Reply:
I agree. I can do this later tonight (I think!).

Reply:
@boyceg can you do this or shall I take a look?